### ISSUE TITLE
HP OV Baseline Error on PowerShell

### ISSUE THREAD

**Issue by HPE-OV** [HewlettPackard/POSH-HPOneView#552]
On OneView v5.00.01 and Power Shell v5.0.2400.1845

Utilizing the PowerShell module HPOneView.500, when the command "Get-HPOVBaseline" is executed to return a list of firmware bundles, the following error is noticed:

Get-HPOVBaseline : Cannot convert the JSON string because a dictionary that was converted from the string contains the duplicated keys 'ri10' and 'RI10'. (Please refer the attachment)

It's because of a custom SPP 2019.09.0 which is being used by one of the Server Profiles.

PowerShell module was updated to version 5.0.2597.2846, and issue is still the same.

![image001](https://user-images.githubusercontent.com/78871738/107548844-23fee900-6bf5-11eb-91f9-1848e7705900.png)

**Comment by ChrisLynchHPE** [comment]
The challenge here is that our SPP's can contain multiple components of the same type.  And how that is then collected and stored is in a Camel Case definition.  PowerShell is not a camel case programming/scripting environment.  Can you please do the following and provide the captured log file:

```powershell
Get-HPOVCommandTrace -ScriptBlock { Send-HPOVRequest -uri /rest/firmware-drivers }
```

Reply with and attach the generated file to this thread so I can analyze it.

**Comment by HPE-OV** [comment]
> The challenge here is that our SPP's can contain multiple components of the same type. And how that is then collected and stored is in a Camel Case definition. PowerShell is not a camel case programming/scripting environment. Can you please do the following and provide the captured log file:
> 
> ```powershell
> Get-HPOVCommandTrace -ScriptBlock { Send-HPOVRequest -uri /rest/firmware-drivers }
> ```
> 
> Reply with and attach the generated file to this thread so I can analyze it.

Here is the Output:

[2102101159_HPOV_transcript.log](https://github.com/HewlettPackard/POSH-HPEOneView/files/5960403/2102101159_HPOV_transcript.log)

**Comment by ChrisLynchHPE** [comment]
Okay, one more:

```powershell
Get-HPOVCommandTrace -ScriptBlock { Send-HPOVRequest -uri "/rest/index/resources?category=firmware-drivers&sort=name:asc" }
```

I think this could be an index resource causing this issue, which this capture would provide.

**Comment by HPE-OV** [comment]
> > The challenge here is that our SPP's can contain multiple components of the same type. And how that is then collected and stored is in a Camel Case definition. PowerShell is not a camel case programming/scripting environment. Can you please do the following and provide the captured log file:
> > ```powershell
> > Get-HPOVCommandTrace -ScriptBlock { Send-HPOVRequest -uri /rest/firmware-drivers }
> > ```
> > 
> > 
> > Reply with and attach the generated file to this thread so I can analyze it.
> 
> Here is the Output:
> 
> [2102101159_HPOV_transcript.log](https://github.com/HewlettPackard/POSH-HPEOneView/files/5960403/2102101159_HPOV_transcript.log)

This is the error he is getting:
![Error2](https://user-images.githubusercontent.com/78871738/107558617-00da3680-6c01-11eb-9532-de67a2b2baba.png)

**Comment by ChrisLynchHPE** [comment]
Please look at my post again.  This is a typo and I fixed that in an edit.  The uri parameter value needs to be encased with `"`:

```powershell
Get-HPOVCommandTrace -ScriptBlock { Send-HPOVRequest -uri "/rest/index/resources?category=firmware-drivers&sort=name:asc" }
```

**Comment by HPE-OV** [comment]
> Please look at my post again. This is a typo and I fixed that in an edit. The uri parameter value needs to be encased with `"`:
> 
> ```powershell
> Get-HPOVCommandTrace -ScriptBlock { Send-HPOVRequest -uri "/rest/index/resources?category=firmware-drivers&sort=name:asc" }
> ```

He got the same error:
![Error3](https://user-images.githubusercontent.com/78871738/107561687-dbe7c280-6c04-11eb-84c9-462cc5fb9a53.png)

**Comment by ChrisLynchHPE** [comment]
Yes, that is expected.  The file created (`2102101313_HPOV_transcript.log`) contains the information I need to see.  Please provide that.

**Comment by HPE-OV** [comment]
> 2102101313_HPOV_transcript.log

Here is the output:
[2102101313_HPOV_transcript.log](https://github.com/HewlettPackard/POSH-HPEOneView/files/5961214/2102101313_HPOV_transcript.log)

**Comment by ChrisLynchHPE** [comment]
Thanks.  I have identified a fix for this, and will be in the next library updates.

**Comment by HPE-OV** [comment]
> Thanks. I have identified a fix for this, and will be in the next library updates.

For PowerShell or OneView?

**Comment by ChrisLynchHPE** [comment]
This is an issue with the PowerShell library, not OneView.  This issue tracker is for the library only, not HPE OneView support.

**Comment by HPE-OV** [comment]
> This is an issue with the PowerShell library, not OneView. This issue tracker is for the library only, not HPE OneView support.

Understood, thank you Chris.

**Comment by ChrisLynchHPE** [comment]
I have just released a 5.00 library update that resolves this issue:

This is fixed in [Release 5.0.2689.2416](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v5.0.2689.2416) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.500/5.0.2689.2416).

I will be releasing updates to the other library versions that also address this.

**Comment by ChrisLynchHPE** [comment]
Closing due to no further activity.  If you wish, feel free to re-open to discuss further.
